pkg/runctl/run_manager_test.go

```diff
@@ -1782,6 +1776,148 @@ func Test_RunManager_Log_Elasticsearch(t *testing.T) {
 			assert.Assert(t, is.Nil(param))
 		})
 	}
+
+	/**
+	 * Test: If provided indexURL at spec.logging.elasticsearch.indexURL
+	 * does not have correct format test should fail.
+	 */
+	test = "ErrorOnWrongFormattedIndexURL"
+	for _, tc := range []struct {
+		name string
+		URL  string
+	}{
+		{"indexURLWithNoScheme", `"indexURL": "testURL"`},
+		{"indexURLWithWrongFormat2", `"indexURL": "http//testURL"`},
+	} {
+		t.Run(test+"_"+tc.name, func(t *testing.T) {
+			// setup
+			pipelineRunJSON := fmt.Sprintf(fixIndent(`
+				{
+					"apiVersion": "steward.sap.com/v1alpha1",
+					"kind": "PipelineRun",
+					"metadata": {
+						"name": "dummy1",
+						"namespace": "namespace1"
+					},
+					"spec": {
+						"jenkinsFile": {
+							"repoUrl": "dummyRepoUrl",
+							"revision": "dummyRevision",
+							"relativePath": "dummyRelativePath"
+						},
+						"logging": {
+							"elasticsearch": {
+								%s
+							}
+						}
+					}
+				}`),
+				tc.URL,
+			)
+			t.Log("input:", pipelineRunJSON)
+			examinee, runCtx, _ := setupExaminee(t, pipelineRunJSON)
+
+			// exercise
+			resultError := examinee.createTektonTaskRun(runCtx)
+
+			// VERIFY
+			assert.ErrorContains(t, resultError, "scheme not supported")
```

Comment:
This does not match to line 1784
Do you only want to test wrong scheme or also other validation errors?
May be also make error configurable in test.
Add more tests.
"foo///bar"
"foo::bar"

Reply:
I merged two tests together now. We only throw "scheme not supported" error in the run_manager with regards to indexURL so I afraid making it configurable could help. I renamed line 1784 to have better name for the test.